src/Nethermind/Nethermind.Facade/Simulate/SimulateBridgeHelper.cs

```diff
@@ -185,7 +185,7 @@ private static BlockHeader GetParent(BlockHeader parent, SimulatePayload<Transac
         ulong lastKnown = (ulong)latestBlockNumber;
         if (firstBlock?.BlockOverrides?.Number > 0 && firstBlock.BlockOverrides?.Number < lastKnown)
         {
-            Block? searchResult = blockTree.FindBlock((long)firstBlock.BlockOverrides.Number);
+            Block? searchResult = blockTree.FindBlock((long)firstBlock.BlockOverrides.Number - 1);
```

Comment:
This is the main issue. Due to a off-by-one error we were not picking the parent but rather the child of the parent. This only happens when we don't use `"latest"` as the block parameter.